src/DockManager.cpp

```diff
@@ -1147,6 +1151,21 @@ CDockFocusController* CDockManager::dockFocusController() const
 	return d->FocusController;
 }
 
+//===========================================================================
+bool CDockManager::loadTranslation(const QString& language)
+{
+    bool result = false;
+
+    // remove the old translator
+    qApp->removeTranslator(d->Translator);
+
+    // load the new translator
+    if (d->Translator->load(language))
+        result = qApp->installTranslator(d->Translator);
+
+    return result;
```

Comment:
There a texts in ADS like in "DockWidgetTab.cpp":

	internal::setToolTip(CloseButton, QObject::tr("Close Tab"));

Is there a way to translate those texts without changing ADS? It was my impression a translation has to be added to the library itself to reach those resources - but I am no expert with the QT translation system - maybe there is another way?

Thanks.

Reply:
Since the translator is loaded in the qApp, you could just create and install the translator, then load your .ts file before CDockManager gets instantiated. It should work.

Reply:
Uh - maybe I am missing something here. How can this work?

It is my understanding there has to be a "ts"-file which is compiled into a QM-file to provide the necessary translations.

The TS-file is generated using the QT command line tools and looks like this:

<context> <name>ads::CDockWidgetTab</name> <message> <location filename="DockWidgetTab.cpp" line="490"/> <source>Detach</source> <translation>Ablösen</translation> </message> </context>

The resulting QM file (containing the translations) can then be loaded at runtime using "QTranslator::load".

How is an application using ADS supposed to know which strings are contained in ADS? How can it provide a "QM"-file containing the translations? What happens if ADS adds new strings?